packages/host/app/services/matrix-service.ts

```diff
@@ -706,6 +709,24 @@ export default class MatrixService extends Service {
     }
   }
 
+  @cached
+  get roomResources() {
+    this.updateRoomResourcesCache();
+    return this.roomResourcesCache;
+  }
+
+  private updateRoomResourcesCache() {
+    for (let roomId of this.rooms.keys()) {
+      if (this.roomResourcesCache.has(roomId)) {
+        continue;
+      }
+      this.roomResourcesCache.set(
+        roomId,
+        getRoom(this, () => roomId),
+      );
+    }
+  }
```

Comment:
in general this is a dangerous pattern that has gotten us into trouble before with the 10,000s of EC tasks from cards grid. is it possible to refactor this so that you do not need to create a resource when this getter is called? let's try to have getters without side effects, and if we need side effects in a getter lets fully document why in the comments.

Reply:
I added a cache to solve that problem so we don't re-create `RoomResource` every time the getter getting triggered.

Reply:
a cache makes it better--but in general getters should not have side effects, and if so it should be well documented so that someone making a modification can understand the implications.

Reply:
also consider the fact you are making a brand new resource each time instead of reusing the one you already have by changing the thunk function.